Source/DafnyCore/AST/Node.cs

```diff
@@ -86,8 +108,19 @@ public abstract class Node : INode {
         return OwnedTokensCache;
       }
 
+      var childrenFiltered = GetConcreteChildren().ToList();
+
+      // DEBUG: Detect duplicate children start tokens.
+      for (var i = 0; i < childrenFiltered.Count - 1; i++) {
```

Comment:
Good idea. Actually, I prefer to remove this code for now, because it allowed me to detect issues in the line that convert children to a dictionary, but these issues are all fixed for now.